### PR TITLE
Create MSCA_application.csl

### PR DESCRIPTION
**Pull request by DaniBodor**
Please see discussion https://forums.zotero.org/discussion/comment/281719 for comments and limitations on this style!
Specifically, the comment of 14 Sept 2016 (https://forums.zotero.org/discussion/comment/262041/#Comment_262041)

### CODE REVIEW

File: MSCA_application.csl [hunk 1]
```diff
@@ -0,0 +1,1256 @@
...
+    <contributor>
+      <name>Andrew Dunning</name>
+      <email>andrew.dunning@utoronto.ca</email>
+    </contributor>
+    <contributor>
+      <name>Simon Kornblith</name>
+      <email>simon@simonster.com</email>
+    </contributor>
+    <contributor>
+      <name>Dani Bodor</name>
```
Comment (POBrien333) [inline MSCA_application.csl]: Make yourself the author and remove the contributors.
Add a template link for the style you based this on as that is necessary to be accepted and also acts as a reference to the authors/contributors for that original style.

File: MSCA_application.csl [hunk 2]
```diff
@@ -0,0 +1,1256 @@
+<?xml version="1.0" encoding="utf-8"?>
+<style xmlns="http://purl.org/net/xbiblio/csl" class="note" version="1.0" demote-non-dropping-particle="never" page-range-format="chicago">
+<!-- This style was edited with the Visual CSL Editor (http://editor.citationstyles.org/visualEditor/) -->
+  <info>
+    <title>Marie Skłodowska-Curie Actions</title>
+    <title-short>Marie Curie fellowship</title-short>
```
Comment (POBrien333) [inline MSCA_application.csl]: remove the title-short.

File: MSCA_application.csl [hunk 3]
```diff
@@ -0,0 +1,1256 @@
...
+    <id>http://www.zotero.org/styles/marie-skłodowska-curie-actions</id>
+    <link href="http://www.zotero.org/styles/marie-skłodowska-curie-actions" rel="self"/>
+    <link href="http://www.chicagomanualofstyle.org/tools_citationguide.html" rel="documentation"/>
```
Comment (POBrien333) [inline MSCA_application.csl]: Is there some proper documentation to this reference style? Or can you upload something at least? This is also important for the naming of the style.

File: MSCA_application.csl [hunk 4]
```diff
@@ -0,0 +1,1256 @@
...
+      <name>Simon Kornblith</name>
+      <email>simon@simonster.com</email>
+    </contributor>
+    <contributor>
+      <name>Dani Bodor</name>
+    </contributor>
+    <category citation-format="note"/>
+    <category field="generic-base"/>
+    <summary>For use in application to H2020 Marie Skłodowska-Curie Actions fellowship</summary>
+    <updated>2016-09-12T17:37:20+00:00</updated>
```
Comment (POBrien333) [inline MSCA_application.csl]: Update this date to something more recent (when you uploaded the style maybe).

File: MSCA_application.csl [hunk 5]
```diff
@@ -0,0 +1,1256 @@
...
+  </macro>
+  <macro name="contributors-short">
+    <group delimiter=" ">
+      <names variable="author">
+        <name form="short" and="symbol"/>
+        <substitute>
+          <names variable="editor"/>
+          <names variable="translator"/>
+        </substitute>
+        <et-al/>
```
Comment (POBrien333) [inline MSCA_application.csl]: remove the et-al line as it's not used anyway. (also should directly be under name and before substitute to get working,if  you were to include it).

File: MSCA_application.csl [hunk 6]
```diff
@@ -0,0 +1,1256 @@
...
+                        <group delimiter=" ">
+                          <text variable="container-title" form="short" strip-periods="true" font-style="italic"/>
+                          <text variable="volume" strip-periods="true" font-weight="bold" suffix=":"/>
+                          <text variable="page" form="short" strip-periods="false"/>
+                        </group>
+                      </else>
+                    </choose>
+                  </group>
+                  <text macro="case-issue-subsequent"/>
+                  <choose/>
```
Comment (POBrien333) [inline MSCA_application.csl]: delete this choose on 1201.

**Comment by DaniBodor** [comment]
There is no proper guideline apart from a line in the application form
stating: "Literature should be listed in footnotes, font size 8 or 9. All
literature references will count towards the page limit." The application
form can only be reached after initiating an application in their system.
I feel that perhaps the most appropriate documentation would be the zotero
forum discussion, so I put the link for that for the moment. Is that
acceptable?

**Comment by POBrien333** [comment]
That then means there is no specific style that MUST be used as is common for grant applications. Why did you make this specific style then and not just use the template style as is?

**Comment by DaniBodor** [comment]
There is indeed no specific style that MUST be used. However, there is no template style anywhere and so I created this as what I feel works best.
According to the comments in the forum, people feel that this would be a useful style for them (or perhaps at least a starting point) and hence I uploaded it like this.

**Comment by DaniBodor** [comment]
I have now made all the changes you suggested. Is there a way I can run it through the checks again? Or is there a different way to proceed?

**Comment by POBrien333** [comment]
It starts the test automatically once you edited the code, but you can always use the validator: http://validator.citationstyles.org/
It also interprets the errors in a bit nicer way, so they're easier to understand.

**Comment by DaniBodor** [comment]
I am currently getting an issue that I don't know how to fix. Any advice?
"Line 1142: Element “choose” from namespace “http://purl.org/net/xbiblio/csl” is missing a required instance of child element “if”."

**Comment by POBrien333** [comment]
Can you update the code on here? It's still your initial commit, so the lines don't match anymore and I have no idea what you're looking at.
I've made comments for the 3 main errors though,s o they should fix everything.

**Comment by DaniBodor** [comment]
There is a new branch called updates_20170731. I was not able to update the code without creating a branch, so this is what I did. Should I have done this differently?
Is this visible to you or is it private? If private, is there any way I can change the privacy setting?
Anyway, here is the link: https://github.com/DaniBodor/styles/commit/4114529c689184432b3e6a82bc85cf79bca4ea3d

Also, do you mean your initial comments? I've fixed all of those. Or have you made new comments now? 
I checked the new file and the only error found is the one I mention above.

Apologies for the confusion, I am very novice at GitHub and CSL editing, etc.

**Comment by POBrien333** [comment]
Yeah, took me ages to get GitHub.
If you're reading this on GitHub, scroll up, click on "Files changed", and there click on the little pencil which lets you edit the file. Copy your new code over and press submit or something like that. :)

**Comment by POBrien333** [comment]
1. File name must match your ID+csl.
2. Remove the following macros as they're not used: ["access-note", "container-contributors-note", "container-title-note", "contributors-note", "contribu...olon", "point-locators-join-with-comma", "secondary-contributors-note", "title-note", "title-short"

File: MSCA_application.csl [hunk 7]
```diff
@@ -0,0 +1,1234 @@
...
+    <choose>
+      <if type="legal_case">
+        <text macro="issue"/>
+      </if>
+    </choose>
+  </macro>
+  <citation et-al-min="3" et-al-use-first="1" disambiguate-add-names="true">
+    <layout suffix="." delimiter="; ">
+      <choose>
+        <else>
```
Comment (POBrien333) [inline MSCA_application.csl]: This is still an issue. I've commented on this already. It needs to start with an if, can't be an else. You must have removed some code before this and hence it's not making sense anymore.
Remove lines 1141,1142 and the matching closing brackets further down 1185+1186.

**Comment by DaniBodor** [comment]
Not sure what you mean by the "contribu...olon" macro

**Comment by POBrien333** [comment]
... is an internationally known space holder. You only got a handful of Macros, I'm sure you can figure out which one starts with contribu and ends with olon. ;)
These all need removing, see the test report: "archive-note", "contributors-sort", "date-disambiguate", "description-note", "editor-translator", "locators-note-join-with-comma", "locators-note-join-with-space", "point-locators-join-with-colon"

You also didn't edit the file name...

**Comment by DaniBodor** [comment]
none of them start with contribu and end with olon. that's why it was unclear. 
Also, are you not able to remove these issues for me, rather than pointing them out?
anyway, i've removed the ones you mentioned.
I didn't change the file name, but instead I changed the ID to match the file name. Is that not good enough?

No errors are detected in the "http://validator.citationstyles.org/" system anymore but apparently there are some here. Will remove more unused macros and see what happens

**Comment by DaniBodor** [comment]
ok, so i am removing unused macros as fast as it is detecting them, will go through more and more rounds of this until it's done.
the other error I keep getting is regarding the filename, which I don't understand what the issue is. Can you tell me what is allowed and what is not.

Thanks!

This is the error:

1) MSCAapplication: must have a conventional file name
     Failure/Error: expect(filename).to match(/^[a-z\d]+(-[a-z\d]+)*\.csl$/)
     
       expected "MSCAapplication.csl" to match /^[a-z\d]+(-[a-z\d]+)*\.csl$/
       Diff:
       @@ -1,2 +1,2 @@
       -/^[a-z\d]+(-[a-z\d]+)*\.csl$/
       +"MSCAapplication.csl"
       
     Shared Example Group: "style" called from ./spec/styles_spec.rb:216
     # ./spec/styles_spec.rb:13:in `block (2 levels) in <top (required)>'

**Comment by DaniBodor** [comment]
ok, and this is now the only issue detected.

**Comment by POBrien333** [comment]
the filenmae must match your ID. so use the ID and add .csl. I had also written that before ;)

**Comment by POBrien333** [comment]
I'm afraid you did it the wrong way round. If you look at the ids and filenames of all the other styles we have in the repository you'll see it is something like marie-skłodowska-curie-actions and marie-skłodowska-curie-actions.csl ;)

**Comment by DaniBodor** [comment]
FYI, the issue was that it only accepts lower case letters

**Comment by DaniBodor** [comment]
What happens now?

**Comment by POBrien333** [comment]
Please change to marie-sklodowska-curie-actions and marie-sklodowska-curie-actions.csl as I had indicated.

**Comment by rmzelle** [comment]
> There is indeed no specific style that MUST be used.

My apologies, but I don't think this style is a good fit for the repository, if there are no citation format requirements beyond having to use a note-style.

(@adam3smith, thoughts? The guidelines for NSF grant proposals are also poorly defined, but at least they define a set required fields; see https://www.zotero.org/styles?q=grant%20proposals and their "documentation" links)

(@POBrien333, I appreciate the help, but for cases like these you might want to wait until we make a decision on whether or not to accept the style, before tidying it up, so that we don't waste the submitter's time in case we don't take it)

**Comment by POBrien333** [comment]
I only helped him/her as there was another user wanting to use the style and it didn't validate.

**Comment by DaniBodor** [comment]
... it worked for me before any of this. 
Anyway, @POBrien333  your help was appreciated!

Also, i don't really care if it goes in the official repository or not. People apparently want to use this style and they can either find it on the repository or here as an unofficial style. It's all the same.

**Comment by rmzelle** [comment]
@DaniBodor, do you know if these grant proposals are always in English?

**Comment by DaniBodor** [comment]
I believe so

**Comment by rmzelle** [comment]
Okay, I fixed the style to (British) English.

I'm closing this now. You can give people a direct link to the style at https://raw.githubusercontent.com/DaniBodor/styles/patch-1/marie-sklodowska-curie-actions.csl (which you can keep updating yourself by using the edit icon at https://github.com/DaniBodor/styles/blob/patch-1/marie-sklodowska-curie-actions.csl).

**Pull request closed without merging**